C++/syntax_test_c.c

```diff
@@ -247,13 +285,329 @@ if (4) {
+/*                         ^^^^^ entity.name.type.typedef.c */
+/*                              ^ punctuation.separator.c */
+                           ull2_t,
+/*                         ^^^^^^ entity.name.type.typedef.c */
+/*                               ^ punctuation.separator.c */
+                           *ullp_t;
+/*                         ^ storage.modifier.c */
+/*                          ^^^^^^ entity.name.type.typedef.c */
+/*                                ^ punctuation.terminator.c */
 
 typedef struct mystruct {
 /* <- storage.type */
-/*             ^ - entity */
+/*      ^^^^^^ storage.type.c */
+/*             ^ entity.name.struct.c */
+/*                      ^ meta.struct.body.c meta.block.c punctuation.section.block.begin.c */
+  int a;
+/*^^^ storage.type.c */
+/*    ^ entity.name.member.c */
 } mystruct;
-/* ^ entity.name.type */
+/* <- meta.struct.body.c meta.block.c punctuation.section.block.end.c */
+/*^^^^^^^^ entity.name.type */
+/*        ^ punctuation.terminator.c */
+
+typedef struct another
+/* <- storage.type */
+/*      ^^^^^^ storage.type.c */
+/*             ^ entity.name.struct.c */
+{
+/* <- meta.struct.body.c meta.block.c punctuation.section.block.begin.c */
+  int f;
+/*^^^ storage.type.c */
+/*    ^ entity.name.member.c */
+} another_t, another2_t;
+/* <- meta.struct.body.c meta.block.c punctuation.section.block.end.c */
+/*^^^^^^^^^ entity.name.type */
+/*         ^ punctuation.separator.c */
+/*           ^^^^^^^^^^ entity.name.type.typedef.c */
+/*                     ^ punctuation.terminator.c */
+
+typedef another_t another2_t;
+/* <- storage.type */
+/*      ^^^^^^^^^ support.type.c */
+/*                ^^^^^^^^^^ entity.name.type.typedef.c */
+/*                          ^ punctuation.terminator.c */
+
+typedef another_t *anotherp_t;
+/* <- storage.type */
+/*      ^^^^^^^^^ support.type.c */
+/*                ^ storage.modifier.c */
+/*                 ^^^^^^^^^^ entity.name.type.typedef.c */
+/*                           ^ punctuation.terminator.c */
+
+typedef void (*xhci_get_quirks_t)(struct device *d, struct xhci_hcd *);
+/*<- storage.type.c */
+/*      ^^^^ storage.type.c */
+/*            ^ storage.modifier.c */
+/*             ^^^^^^^^^^^^^^^^^ entity.name.type.c */
+/*                                ^^^^^^ storage.type.c */
+/*                                       ^^^^^^ support.type.c */
+/*                                              ^ storage.modifier.c */
+/*                                               ^ variable.parameter.c */
+/*                                                ^ punctuation.separator */
+/*                                                  ^^^^^^ storage.type.c */
+/*                                                         ^^^^^^^^ support.type.c */
+/*                                                                  ^ storage.modifier.c */
+
+typedef struct Node* (*get_node_t)(struct device *d, struct xhci_hcd *);
```

Comment:
I like the `storage.modifier` here. Is there precedent for the scope specifically, or did you pick it?

Reply:
Which instance specifically? The prototype name? I guess I picked that one. It's not really modifying the data type, but it's required to declare it. So maybe it shouldn't be `storage.modifer`?

Reply:
Ah, you are referring to the `*`.
I picked it because of description:
> Keywords that affect the storage of a variable, function or data structure should use the following scope.